code/modules/hydroponics/seeds.dm

```diff
@@ -302,6 +303,14 @@
 			T.reagents.handle_reactions()
 			playsound(T.loc, 'sound/effects/space_wind.ogg', 50)
 
+/// Returns reagent purity based on seed stats
+/obj/item/seeds/proc/get_reagent_purity()
+	var/purity_from_lifespan = lifespan / 400 //up to +25% for lifespan
+	var/purity_from_endurance = endurance / 400 //up to +25% for endurance
+	var/purity_from_instability = rand(-instability, instability) / 400  //up to +-25% at random for instability
+	var/result_purity = clamp(0.5 + purity_from_lifespan + purity_from_endurance + purity_from_instability, 0, 1) //50% base + stats
```

Comment:
Should probably either increase the bonuses or add a new factor, since even with maximum lifespan and endurance, if there's _any_ instability you've got a 50% chance of your produce not being at 100% purity which is kind of disappointing.

Reply:
it will be only +-2.5% of purity if you have 10 instability. You can easily remove it with robust harvest or diethylamine. And purity is an invisible stat (you can only figure out the exact value in chromatography machine) so reaching 100% purity is not the goal. It may even be possible for admin spawn stuff only, or if you know how to cross-pollinate stuff onto plants with 100 lifespan.

Reply:
You can actually find purity with the Chemical analyzer, which is handheld.

Reply:
Well anyway, 98% pure reagents will mean that you get the 10 minute buff reduced by like 12 seconds. And they will be only 2% less filling, which is negligible. The effects are scaled, so 100 is just a cap, rather than a goal.